### PR TITLE
#167278986 Builds endpoint Register new User

### PR DESCRIPTION
**Pull request by b0nbon1**
#### What does this PR do?
- Builds endpoint Register new user
#### Description of Task to be completed?
- write failing tests
- configure tests with db
- set up connection with database
- write the endpoint register user
- validate the users details
- make sure the tests are passing
#### How should this be manually tested?
- git clone the repo
- cd into the repo
- to start the application run `npm start`
- go to postman and test endpoint `api/v1/auth.signup`
- run the tests using `npm test` on your terminal
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#167278986](https://www.pivotaltracker.com/n/projects/2353886/stories/167278986)
#### Screenshots:
Screenshot from 2019-07-16 08-52-56

#### Questions:
- how can we store the token returned to avoid being stolen?

### CODE REVIEW

File: src/database/index.js
Comment (Kasulejoseph) [inline src/database/index.js]: Don't merge the `console.log`, remove all of them before merging

**Review by Kasulejoseph — CHANGES_REQUESTED** [review]

File: src/routes/index.js
Comment (Kasulejoseph) [inline src/routes/index.js]: This shouldn't be 405, let's have a 2 minutes sync and show you why.

File: src/database/index.js
Comment (b0nbon1) [reply src/database/index.js]: Oh okay okay. Thanks
I fixed that

File: src/routes/index.js
Comment (b0nbon1) [reply src/routes/index.js]: Fixed it, handled both the 405 and the 404
Thanks for that

**Pull request merged**
